sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/util/TreeUtil.kt

```diff
@@ -231,6 +231,7 @@ private fun ArrayList<SqlCreateTableStmt>.buildGraph(): Graph<SqlCreateTableStmt
 
 private fun <V, E> Graph<V, E>.topological(): Iterator<V> = TopologicalOrderIterator(this)
 
+// TODO Consider lifting into Dialect class
```

Comment:
In doing so, we make the compiler less reliant on knowing about dialect capabilities, but make dialect implementations more aware of compiler inner workings. If this change is desired, I'm happy to make it, and in the process, make the `Dialect` subclasses for each dialect. Ideally as a separate PR IMO.

Reply:
any hard references to dialects in the compiler will eventually need to leave, so we should very liberally move stuff into the dialect API in cases like that. It might be worth revisiting the API in general and if we can avoid it, but for now I think moving it makes sense.